### DISCUSSION TITLE
Add Language to 'grammarly' LSP (Configuring LSPs)

### DISCUSSION THREAD

**Discussion by h0useofdupree** [williamboman/nvim-lsp-installer#497] [Q&A] [answered]
### Hello fellow people

I can imagine that this is self-explanatory and I might say that I have held back asking such question. But I am at the end of my knowledge so I have to ask.

`If there is a way, how may I add a language to the 'grammarly' server. Basically I am asking for general info on configuring LSPs installed in nvim. `

Thanks in advance!

**Cheers**

**Comment by williamboman** [comment, marked as answer]
Hey! So the way neovim's LSP work (thanks to the lspconfig plugin, which is not a builtin mechanism but instead is a plugin and is subject to change) is that it starts and attaches servers to (neo)vim buffers depending on 1) their `'filetype'` (`:h 'filetype'`) and 2) whether it's able to detect a root directory (see `:h lspconfig-root-dir`).

The grammarly server is configured to start and attach to [these](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/grammarly.lua#L13) filetypes, which currently only is `markdown`. If you want to extend these filetypes while using nvim-lsp-installer I'd recommend something like:

```lua
local lsp_installer = require("nvim-lsp-installer")

lsp_installer.on_server_ready(function (server)
    local opts = {}
    if (server.name == "grammarly") then
        opts.filetypes = { "markdown", "rst", "html" ] -- add the filetypes you want grammarly to start on here
    end
    server:setup(opts)
end)
```

**Reply by h0useofdupree** [reply]
Hello
first of all thanks for your quick reply. I feel stupid not having said that I wanted to add a language in terms of grammar correction (which grammarly itself supports). In my case that would be german. 

BUT, your reply actually did help, since that is also something I wanted to know. 

I am really sorry for stating my question the wrong way. 

I have since used the ltex extension for md, since with that I am able to set the language in the document itself. 

As said, thank you so much for this! I will implement this right away for other file types I write in. 


Cheers

**Reply by williamboman** [reply]
Ah! I'm not sure if the grammarly server supports more languages than just English (default) at the moment. I'm sure you could find more info over at https://github.com/znck/grammarly

**Reply by h0useofdupree** [reply]
Oh man you are right! Don't know what I missed. Thanks for the tip though, for now I'll be using ltex. 

Have a great day/night. :)
